### PR TITLE
Expose as multi-purpose library

### PR DESCRIPTION
**Pull request by evansolomon**
Hello @dpup, 

Please review the following commits I made in branch 'expose-as-obj'.

fcd0c84804065a21e383cb85e0a829b4dfc9a746 (2013-08-27 17:37:28 -0700)
Build docs

102a7983803e90a73fdfdf0a51c6f1acaec5d1be (2013-08-27 17:36:57 -0700)
Update name and readme for new API

40c0b07c45d211e7991244154026876d50a5144d (2013-08-27 17:36:33 -0700)
Use new API in unit tests
Green!

cecd9cc4ae2686bb98b99219ed5d5c43a44a627a (2013-08-27 17:35:49 -0700)
Expose a constructor instead of a single method
Prepare for a multi-purpose library

R=@dpup

### CODE REVIEW

**Comment by evansolomon** [comment]
I renamed it to "S3 Utils" for now. I'd like to come up with a more creative name though.

File: index.js [hunk 1]
```diff
@@ -1,5 +1,23 @@
 // Copyright 2013 The Obvious Corporation.
 
+// ## S3 Utils
```
Comment (dpup) [inline index.js]: These top level blocks should be `/** */ style`

File: index.js [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+//
+// Usage:
+//
+// ```
+// var AWS = require('aws-sdk');
+// var s3 = new AWS.S3(configParams);
+// var s3Utils = new S3Utils(s3);
+// ```
+
+var S3Utils = module.exports = function(s3) {
```
Comment (dpup) [inline index.js]: I prefer:

```
function S3Utils(s3sdk) {

}
module.exports = S3Utils
```

File: index.js [hunk 3]
```diff
@@ -10,17 +28,13 @@ var S3Stream = require('./lib/Stream');
...
 // ```
-// var AWS = require('aws-sdk');
-// var s3 = new AWS.S3(configParams);
-// var createS3WriteStream = require('s3-write-stream');
-// var targetFile = {Bucket: 'random-access-memories', Key: 'to-get-lucky.log'};
-// var s3stream = createS3WriteStream(s3, targetFile);;
+// var s3stream = s3Utils.createWriteStream({
+//   Bucket: 'random-access-memories',
+//   Key: 'to-get-lucky.log'
+// });;
```
Comment (dpup) [inline index.js]: `;;`

We should really make sure all our JS is the same style, but that can be done as a followup.

File: index.js [hunk 4]
```diff
@@ -33,10 +47,10 @@ var S3Stream = require('./lib/Stream');
 //   when the stream is ready
 // * Start writing immediately and respect `false` return values. This is how
 //   Node's `stream.pipe()` behaves
-module.exports = function(s3, params, callback) {
-  var s3stream = new S3Stream(params, s3);
+S3Utils.prototype.createWriteStream = function(params, callback) {
```
Comment (dpup) [inline index.js]: Space after function:

`function (params, callback) {`

**Comment by dpup** [comment]
LGTM

File: index.js (hunk 1 above)
Comment (evansolomon) [inline index.js]: Docco can't parse multi-line comments (and [probably never will](https://github.com/jashkenas/docco/pull/63#issuecomment-11504831)) :(

Obviously Docco documentation isn't the most important thing, but it is a nice way to make decent looking docs.  What do you think, better to standardize style and lose/change the docs?

Comment (dpup) [inline index.js]: eh, my preference is to standardize and use a doc parser that understands
JsDoc and JS Compiler types.  I mostly use code as the reference anyway.

-- Dan

**Pull request merged**